### PR TITLE
Add doc on setting up a dev environment

### PR DESCRIPTION
**Pull request by 2DSharp**
Addressing https://github.com/jupyter/qtconsole/issues/302 for beginner friendly contributing page to set up development environments.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,3 +1,31 @@
...
+```
+Setup **qtconsole** to run on the development environment:
+
+```
+  python setup.py develop
+```
+
+To run after the changes have been made to source:
+```
+  pip install -e .
```
Comment (wmvanvliet) [inline CONTRIBUTING.md]: What's the difference between `python setup.py develop` and `pip install -e .`? Why should we need to run both? (Hey, I'm learning something new about python, cool!)

Comment (2DSharp) [reply CONTRIBUTING.md]: I don't think you need to run both. Maybe that should be reworded to clear out that confusion. I am not very sure about the difference, but I based my commit on what I understood by Carol's comments: https://github.com/jupyter/qtconsole/issues/302#issuecomment-427733177.

Please correct me if I am wrong.

Comment (2DSharp) [reply CONTRIBUTING.md]: Seems like there are some subtle differences:
https://stackoverflow.com/questions/30306099/pip-install-editable-vs-python-setup-py-develop

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+
+Use Conda to install dependencies and activate the development environment.
+
+```
+conda create -n qtdev python=3
+conda activate qtdev
+conda install pyqt
+```
+
+To run after the changes have been made to source (preferred):
```
Comment (ccordoba12) [inline CONTRIBUTING.md]: Please add blank below this line

Comment (2DSharp) [reply CONTRIBUTING.md]: @ccordoba12 Did you mean after " To run after the changes have been made to source (preferred): "?

Comment (ccordoba12) [reply CONTRIBUTING.md]: Yep, that's what I meant.

Comment (2DSharp) [reply CONTRIBUTING.md]: Added the line and reworded. Let me know if this is what you wanted.

**Comment by 2DSharp** [comment]
@wmvanvliet Is there something I should change about this patch?

File: CONTRIBUTING.md (hunk 1 above)
Comment (wmvanvliet) [reply CONTRIBUTING.md]: Ah, thanks for the link.

**Comment by 2DSharp** [comment]
Thank you! :) I hope this will clear up some confusion for future contributors.

**Comment by ccordoba12** [comment]
@wmvanvliet, agreed, so merging. Thanks @2DSharp!

**Pull request merged**